src/dataflow/src/source/timestamp.rs

```diff
@@ -71,17 +71,21 @@ impl TimestampProposer {
         }
     }
 
-    /// Attempt to propose that `(partition, offset)` be bound to `time`, which means
-    /// that all offsets < `offset` get bound to `time` for `partition`.
+    /// Propose that `(partition, offset)` be bound to whatever timestamp we are
+    /// currently assigning data to.
     ///
-    /// This proposal will be ignored if the `time` does not match the current `time`
-    /// this proposer is operating at, and also if another reader has already proposed
-    /// a binding for an offset greater than `offset`. The only exception here is if
-    /// `time` is 0, which is accepted to bootstrap the timestamp proposal.
+    /// This proposal is ignored if there is already a proposed binding for this
+    /// partition to an offset > 'offset'.
     fn propose_binding(&mut self, partition: PartitionId, offset: MzOffset) -> Timestamp {
-        let next_offset = self.bindings.entry(partition).or_insert(offset + 1);
-        if offset + 1 > *next_offset {
-            *next_offset = offset + 1;
+        // Propose one past the current offset, as bindings store one past the
+        // maximum offset bound to that time..
+        let next_offset = offset + 1;
```

Comment:
These words are more confusing, as the method is called `propose_binding` that takes an offset, and now we say we will "propose one past". What afaict the method does is ensure that an offset is bound to a timestamp. I might call it `ensure_timestamped(partition, offset)`. Then the internal machinery of what a binding actually contains doesn't leak out. However, I would likely say something more like "Use a binding containing *one past* `offset`, to ensure that `offset` is bound to a time"

Reply:
changed the wording to match what you suggested in the last sentence. I _think_ ensure_timestamped makes more sense but I want to think about it a bit more